### PR TITLE
[issue_13] add breakpoints for nodes that represent a relationship

### PR DESCRIPTION
**Pull request by meretp**
This PR adds some functionality to add breakpoints in the opossum file. Each folder representing a relationship should be a brekapoint.

This is a subtask of #13.

### CODE REVIEW

File: src/opossum_lib/file_generation.py [hunk 1]
```diff
@@ -4,21 +4,40 @@
...
+OpossumInformation = TypedDict(
+    "OpossumInformation",
+    {
+        "metadata": Dict[str, str],
+        "resources": Dict[str, Any],
+        "externalAttributions": Dict,
+        "resourcesToAttributions": Dict,
+        "attributionBreakpoints": List[str],
+    },
+)
```
Comment (JakobSchubert) [inline src/opossum_lib/file_generation.py:25]: Is there a special reason you are using this functional syntax to define the typed dict? Looking at https://docs.python.org/3/library/typing.html#typing.TypedDict it seems to me that the other syntax is more common.

File: src/opossum_lib/file_generation.py [hunk 2]
```diff
@@ -53,3 +76,17 @@ def tree_to_dict(graph: DiGraph, source: str) -> Union[int, Dict[str, Any]]:
         successor_name = graph.nodes[successor]["label"]
         branch_dict[successor_name] = tree_to_dict(graph, successor)
     return branch_dict
+
+
+def get_breakpoints(graph: DiGraph, source: str) -> List[str]:
```
Comment (JakobSchubert) [inline src/opossum_lib/file_generation.py:81]: The `graph` is expected to be a tree, right? Maybe rename the parameter `graph` to `tree` to make this clear.

**Review by JakobSchubert — APPROVED** [review]
Nice :-).

File: src/opossum_lib/file_generation.py (hunk 1 above)
Comment (meretp) [reply src/opossum_lib/file_generation.py:25]: No special reason, I didn't work with `TypedDict` before and iirc this was just the result of my google search on a mypy error I had. I will change this to use the more common syntax.

File: src/opossum_lib/file_generation.py (hunk 2 above)
Comment (meretp) [reply src/opossum_lib/file_generation.py]: Correct, I renamed this, also in `tree_to_dict` above.

**Pull request merged**